Assets/Controllers/GameManager.cs

```diff
@@ -153,4 +179,36 @@ public Vector3 GetPawnPosition(Pawn pawn)
         Vector3 pawnPosition = pawnKeeper.GetPawnGOFromPawn(pawn).transform.position;
         return pawnPosition;
     }
+
+    public void EndTurn()
+    {
+        int playerCount = players.Count;
+
+        int playerIndex = players.FindIndex( x => x == activePlayer);
+
+        if(playerIndex + 1 == playerCount)
+        {
+            playerIndex = 0;
+        } else
+        {
+            playerIndex++;
+        }
+        ClearSelectedPawn();
```

Comment:
This looks like a series of teardown or reset statements? I'd abstract that into a testable method OR abstract the index calculation.